File: codegen/results/v002/m3-max-48gb/ollama/gemma4_26b_think/20260411_184609/python/008_prime_numbers/solution.py
import sys

def find_primes(n):
    """
    Uses the Sieve of Eratosthenes to find all primes up to and including n.
    """
    if n < 2:
        return []

    # Create a boolean array "is_prime[0..n]" and initialize
    # all entries it as true. A value in is_prime[i] will
    # finally be false if i is Not a prime, else true.
    is_prime = [True] * (n + 1)
    is_prime[0] = is_prime[1] = False

    for p in range(2, int(n**0.5) + 1):
        # If is_prime[p] is not changed, then it is a prime
        if is_prime[p]:
            # Updating all multiples of p starting from p*p
            for i in range(p * p, n + 1, p):
                is_prime[i] = False

    # Return list of all indices that are still marked as True
    return [p for p, prime in enumerate(is_prime) if prime]

def main():
    # Check if the argument was provided
    if len(sys.argv) != 2:
        return

    try:
        # Parse the command line argument to an integer
        n = int(sys.argv[1])
        
        # Calculate primes
        primes = find_primes(n)
        
        # Print each prime on a new line
        for prime in primes:
            print(prime)
            
    except ValueError:
        # If the argument is not an integer, we exit silently 
        # or handle as per requirement.
        pass

if __name__ == "__main__":
    main()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_26b_think/20260411_184609/go/008_prime_numbers/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

// main is the entry point of the program.
func main() {
	// Check if an argument was provided.
	if len(os.Args) < 2 {
		return
	}

	// Parse the argument from string to integer.
	arg := os.Args[1]
	n, err := strconv.Atoi(arg)
	if err != nil {
		// If the argument is not a valid integer, we exit.
		return
	}

	// If N is less than 2, there are no prime numbers to print.
	if n < 2 {
		return
	}

	// We use the Sieve of Eratosthenes algorithm to find all primes up to N.
	// Create a boolean slice "isPrime" and initialize all entries to true.
	// A value in isPrime[i] will finally be false if i is Not a prime, else true.
	isPrime := make([]bool, n+1)
	for i := 2; i <= n; i++ {
		isPrime[i] = true
	}

	// The sieve logic:
	// Start from the first prime number, 2.
	for p := 2; p*p <= n; p++ {
		// If isPrime[p] is not changed, then it is a prime.
		if isPrime[p] {
			// Update all multiples of p starting from p*p.
			// We start at p*p because smaller multiples would have already been marked.
			for i := p * p; i <= n; i += p {
				isPrime[i] = false
			}
		}
	}

	// Iterate through the slice and print all indices marked as true.
	for p := 2; p <= n; p++ {
		if isPrime[p] {
			fmt.Println(p)
		}
	}
}